assets/helpers/checkoutForm/checkoutForm.js

```diff
@@ -87,15 +84,22 @@ export const formElementIsValid = (formElement: Object | null) => {
 
 export const formIsValid = (formClassName: string) => formElementIsValid(getForm(formClassName));
 
-export function checkoutFormShouldSubmit(
-  contributionType: Contrib,
-  isSignedIn: boolean,
+type CheckoutFormShouldSubmitParameters = {|
   isRecurringContributor: boolean,
-  userTypeFromIdentityResponse: UserTypeFromIdentityResponse,
-  form: Object | null,
-) {
-  return formElementIsValid(form)
-    && canContributeWithoutSigningIn(contributionType, isSignedIn, userTypeFromIdentityResponse)
+  canContributeWithoutSigningIn: boolean,
+  formValid: boolean,
+|}
+
+
+export function checkoutFormShouldSubmit(params: CheckoutFormShouldSubmitParameters) {
```

Comment:
This function is now the one place where we decide whether the form should submittable or not.

Reply:
since this function is now beautifully concise and only called in one place, you could simply inline it in `enableOrDisableForm`, hopefully discouraging people from calling it themselves in new places